lib/pages/other/WebView.dart

```diff
@@ -7,37 +9,62 @@ import 'package:app/assets/components/customAppBar.dart';
 import 'package:stacked/stacked.dart';
 import 'package:app/viewmodels/web_view_model.dart';
 
-class WebViewArgumnets {
+import '../../locator.dart';
+
+class WebViewArguments {
   final String initialUrl;
 
-  WebViewArgumnets(this.initialUrl);
+  WebViewArguments(this.initialUrl);
 }
 
-class WebViewPage extends StatelessWidget {
-  final WebViewArgumnets args;
+class WebViewPage extends StatefulWidget {
+  final WebViewArguments args;
   WebViewPage(this.args);
+  
+
+  @override
+  _WebViewPageState createState() => _WebViewPageState();
+}
+
+class _WebViewPageState extends State<WebViewPage> {
+
+ final NavigationService _navigationService = locator<NavigationService>();
 
   @override
   Widget build(BuildContext context) {
     return Scaffold(
-      appBar: CustomAppBar(
+      appBar: customAppBar(
         backButtonText: "Back",
         context: context,
+        actions: <Widget>[
+         TextButton(
+           child: Icon(Icons.more_vert, color: Theme.of(context).primaryColor),
+           onPressed: ((){
+               _navigationService.launchLink(
+                widget.args.initialUrl,
+                isExternal: true,
+                title: 'Open with an External Browser?',
+                description: 'You are about to open this link with an external browser.',
+                buttonText: 'Yes, Go Ahead',
+                closeButtonText: 'Stay Here'
+                );
```

Comment:
Hi @JElgar, Thanks for the feedback but Isn't it correct to make it ` model.launchExternal(widget.args.initialUrl);`?

Reply:
With the ViewModel as it is now yes. However, I think it would make sense if we instead store the `initalUrl` in the ViewModel. This means the ViewModel would already know the URL that the page was launched on and would be able to open it in the browser.

We can do this by adding the `intialUrl` as an argument to the view model constructor and then adding that into the `viewModelBuilder` attribute of the `ViewModelBuilder`.

Reply:
One other thing, that is completely my fault, is I don't think WebView needs to be stateful anymore?

Reply:
Ok.